components/raftstore-v2/src/operation/ready/mod.rs

```diff
@@ -139,14 +198,18 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
             // it may block for ever when there is unapplied conf change.
             self.set_has_ready();
         }
-        if committed_index > applied_index + PAUSE_FOR_RECOVERY_GAP {
+        if committed_index > applied_index + PAUSE_FOR_RECOVERY_GAP && let Some(w) = watch {
             // If there are too many the missing logs, we need to skip ticking otherwise
             // it may block the raftstore thread for a long time in reading logs for
             // election timeout.
             info!(self.logger, "pause for recovery"; "applied" => applied_index, "committed" => committed_index);
-            self.set_pause_for_recovery(true);
+            w.record_paused();
+            self.set_replay_watch(Some(w));
             true
         } else {
+            if let Some(w) = watch {
+                w.record_skipped();
```

Comment:
Why is it called "skipped"?

Reply:
It's "skipped" the pause for replay as it's not needed.

Reply:
It is confusing. For "pause", it means the replay is paused. But for "skipped", it doesn't mean replay is skipped, it means "pause-for-replay" is skipped.

Reply:
Good point. It's renamed to "normal_peers" and "paused_peers", which seems better. @tabokie